data/transactions/logic/eval.go

```diff
@@ -4025,48 +4019,70 @@ func (cx *EvalContext) accountReference(account stackValue) (basics.Address, uin
 	if err != nil {
 		return addr, 0, err
 	}
+
 	idx, err := cx.txn.Txn.IndexByAddress(addr, cx.txn.Txn.Sender)
+	if err == nil {
+		return addr, idx, nil
+	}
+
+	ok := cx.availableAccount(addr)
+	if !ok {
+		return addr, 0, err
```

Comment:
nit:
```suggestion
		return addr, 0, nil
```

Reply:
The `err` is correct.  If there was no error, we have already returned, along with the index of the account that was asked for. Now, we're trying to decide if we should return successfully, but with a sentinel value to indicate that the account is available, but is not accessible by an index into txn.Accounts.

If we have `!ok`, then the account is not accessible at all, so we want to return a failure. And, in fact, the failure is exactly what's already been reported in `err` - the account is not available.

I will add comments to make it clearer, because it took me a while to remember how it works too. (the last paragraph of the comment above the function explains it a little).